src/services/request-service.ts

```diff
@@ -0,0 +1,75 @@
+import Promise = require('bluebird');
+import r = require('rethinkdb');
+import uuid = require('uuid');
+import EmailService = require('./email-service');
+import nodemailer = require('nodemailer');
+import q = require('../dbutils/query');
+import models = require('../models/models');
+import errors = require('../errors/errors');
+import config = require('../config');
+
+
+module RequestService {
```

Comment:
I think this is good for now. Most of the race conditions in other services are not going to happen because a user cannot be deleted in version 1 of the product. I will work on a hotfix to get other db writes race-free.